### DISCUSSION TITLE
sel() using a Coordinate and not a Dimension

### DISCUSSION THREAD

**Discussion by DeSanz** [pydata/xarray#9672] [Q&A] [answered]
Hello,

I have a dataset with the following dimensions and coordinates:

```
Dimensions:                         (current: 20, reps: 3, field_n: 30)
Coordinates:
  * current                         (current) float64 160B 0.0 0.005263 ... 0.1
  * reps                            (reps) float64 24B 0.0 15.0 30.0
  * field_n                         (field_n) int32 120B 0 1 2 3 ... 26 27 28 29
  * field                           (field_n) float32 120B 1.0 2.0 ... -2.0 -1.0
```

and I would like to use the sel() method using as reference the 'field' coordinate (linked to the 'field_n' dimension) to filter a dataset. However when I try to do, for example:
`ds['data_test'].sel(field=[2]) `
I get the following error:

_Dimensions {'field'} do not exist. Expected one or more of ('current', 'field_n')_

I confirmed that selecting using 'field_n' works perfectly: `ds['data_test'].sel(field_n=[2]) `

I am guessing that using sel() for coordinates and not dimensions is not supported or requires some extra step?

Thank you a lot in advace for your help, and thanks for the amazing work in developing xarray!!

**Comment by keewis** [comment, marked as answer]
`isel` only works on dimensions (anything in `ds.dims`), and is used for position-based indexing.

If you truly want label-based indexing (it looks like you created an index for `field_n`), I think you'll have to use `.sel` (i.e. maybe this is a typo?).

**Reply by DeSanz** [reply]
yes, sorry, I want to use sel() it was a typo

**Reply by DeSanz** [reply]
In fact, the problem was finally that the coordinate has repeated values, this is a different issue, thank you for the help
